scripts/xdist/setup_worker.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# Set up worker node.
+
+while getopts 'p:d:' opt; do
+    case "$opt" in
+	p) PYTHON_VERSION="$OPTARG";;
+	d) DJANGO_REQUIREMENT="$OPTARG";;
+	[?])
+	    print >&2 "Usage: $0 -p python-version -d django-reqs-file"
+	    exit 1
+            ;;
+    esac
+done
+
+venv_parent=/home/jenkins/edx-venv-${PYTHON_VERSION}
+venv=$venv_parent/edx-venv
+rm -rf $venv
+tar -C $venv_parent -xf /home/jenkins/edx-venv_clean-${PYTHON_VERSION}.tar.gz
+source $venv/bin/activate
```

Comment:
I can never remember if the environment variables initialized this way actually propagate to the environment that called this script (they need to here).

Reply:
Ah right, because this is now being sourced from a script. Hmm... it looks like maybe some of our code in paver is actually taking care of activating the venv before running the xdist workers: https://github.com/edx/edx-platform/blob/5de9eed420f0e84db1383403f932e48e98c89df8/pavelib/utils/test/suites/pytest_suite.py#L321 (and on another line as well)